### ISSUE TITLE
Reliability of real label calculation method

### ISSUE THREAD

**Issue by Rick-Yqh** [Jiaxin-Lu/Jigsaw#14]
Hello, through the code found that your proposed method of marking real tags is to distinguish real breakpoints from non-breakpoints by comparing the distance between different parts of the points and the size of the threshold n, but how do you ensure that the labels you mark in this method are correct and correspond to the real situation? After all, your subsequent segmentation loss function is based on this real label, if the label calculated by your method is not necessarily correct, it will affect the entire assembly result, and forget that you can answer the doubts, thank you very much!

**Comment by Jiaxin-Lu** [comment]
We assume the ground truth point cloud (gt_pcs) is the correctly assembled object which we can get from the dataset (at least for all datasets in fracture assembly task). Therefore, the computed labels based on this assumption are always correct. Could you elaborate what situation you are facing that makes it wrong?

**Comment by Rick-Yqh** [comment]
No more questions. Thank you very much for your reply. At present, there is still a small problem, that is, why the mat_loss in the training process image provided by you (as shown below) is still very large at the end of the training, and the value of mat_f1 is very small, only 0.08. This does not mean that the predicted mat and the real mat are very different.

![Image](https://github.com/user-attachments/assets/f95e2dbc-49e8-4f9c-ab60-16bd022517b6)

**Comment by Jiaxin-Lu** [comment]
The GT computed here is very strict, so it's natural that the loss and f1 seems bad. Deep learning lesson one should tell you that the absolute value of a metric could be very deceptive.

**Comment by Rick-Yqh** [comment]
Ok, thank you very much for your answer. However, I still have a doubt. Since the indicators of mat_loss and mat_f1 may be very deceptive, how to judge the performance of the obtained mat matrix? Is it possible that the final stitching result is poor is caused by the mat matrix.

**Comment by Rick-Yqh** [comment]
Hello, I would like to ask why the eval evaluation of data after training the model found that cls_f1 and cls_acc were both 1 but cls_loss was not 0.
